File: Webapp/Api/PC_Web_Shop/Endpoints/ArtikalEndpoints/GetAll/ArtikalGetAllEndpoint.cs
using Microsoft.AspNetCore.Mvc;
using Microsoft.EntityFrameworkCore;
using PC_Web_Shop.Data;
using PC_Web_Shop.Helper;

namespace PC_Web_Shop.Endpoints.ArtikalEndpoints.GetAll
{
    [Route("artikal")]
    public class ArtikalGetAllEndpoint : MyBaseEndpoint<ArtikalGetAllRequest,ArtikalGetAllResponse>
    {
        private readonly ApplicationDbContext _applicationDbContext;
        public ArtikalGetAllEndpoint(ApplicationDbContext applicationDbContext)
        {
            _applicationDbContext = applicationDbContext;
        }

        [HttpGet("get-all")]
        public override async Task<ArtikalGetAllResponse> Obradi([FromQuery] ArtikalGetAllRequest request,
            CancellationToken cancellationToken)
        {
            var artikal = await _applicationDbContext.Artikal.Where(x=>!x.IsDeleted)
                
                .OrderByDescending(x => x.Id)
                .Select(x => new ArtikalGetAllResponseArtikal()
                {
                    Id = x.Id,
                    Naziv = x.Naziv,
                    Cijena = x.Cijena,
                    Opis = x.Opis,
                    KratkiOpis = x.KratkiOpis,
                    Model = x.Model,
                    Sifra = x.Sifra,
                    StanjeNaSkladistu = x.StanjeNaSkladistu,
                    Popust = x.Popust,
                    Proizvodjac = x.Prozivodjac,
                    ArtikalKategorija = x.ArtikalKategorija,
                    SlikaArtikla = x.SlikaArtikla
                    
                })
                .ToListAsync(cancellationToken: cancellationToken);

            return new ArtikalGetAllResponse
            {
                Artikal = artikal
            };

        }
    }
}

File: Webapp/Front-end/src/app/endpoints/artikal-endpoints/artikal-getall.endpoint.ts
import {MyBaseEndpoint} from "../MyBaseEndpoint";
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {MojConfig} from "../../moj-config";
import {Injectable} from "@angular/core";
@Injectable({providedIn: 'root'})
export class ArtikalGetallEndpoint implements MyBaseEndpoint<void, ArtikalGetAllResponse>{
  constructor(public httpClient:HttpClient) {
  }
  obradi(request: void): Observable<ArtikalGetAllResponse> {
      let url = MojConfig.adresa_servera+'/artikal/get-all';

    return this.httpClient.get<ArtikalGetAllResponse>(url);
  }

}
export interface ArtikalGetAllResponse{
  artikal:ArtikalGetAllResponseArtikli[];
}
export interface ArtikalGetAllResponseArtikli {
  id: number
  naziv: string
  cijena: number
  opis: string
  kratkiOpis: string
  stanjeNaSkladistu: number
  slikaArtikla: string
  sifra: number
  model: string
  popust: any
  proizvodjac: any
  artikalKategorija: any
  skladiste: any
}
